src/js/common/helpers/login-helpers.js

```diff
@@ -102,25 +102,30 @@ export function addEvent(element, eventName, callback) {
   }
 }
 
-export function getLoginUrls(onUpdateLoginUrls) {
+export function getLoginUrls(dispatch) {
   const loginUrlsRequest = fetch(`${environment.API_URL}/v0/sessions/authn_urls`, {
     method: 'GET',
   }).then(response => {
-    return response.json();
+    if (response.ok) {
+      return response.json();
+    }
+    throw Error(response.statusText);
   }).then(json => {
-    onUpdateLoginUrls(json);
+    dispatch(updateLogInUrls(json));
+  }).catch(() => {
+    dispatch({ type: FETCH_LOGIN_URLS_FAILED });
   });
 
   return loginUrlsRequest;
 }
 
-export function handleLogin(loginUrl, onUpdateLoginUrls) {
+export function handleLogin(loginUrl) {
   window.dataLayer.push({ event: 'login-link-clicked' });
   if (loginUrl) {
     window.dataLayer.push({ event: 'login-link-opened' });
     const receiver = window.open(`${loginUrl}&op=signin`, 'signinPopup', 'resizable=yes,scrollbars=1,top=50,left=500,width=500,height=750');
     receiver.focus();
-    return getLoginUrls(onUpdateLoginUrls || updateLogInUrls);
+  } else {
+    Promise.reject('Could not log in; loginUrl not provided.');
```

Comment:
Shouldn't this be returned? Actually, from a quick glance, none of our uses of `handleLogin` seem to expect a return value except for LoginModal, and it's not even used there correctly because it's only for aborting a request upon unmounting the component. So I'm not sure this `Promise.reject` would do anything.

https://github.com/department-of-veterans-affairs/vets-website/blob/cc4f3019eceba107269f7d51ab58b7ee599f0a9e/src/js/common/components/LoginModal.jsx#L74